lib/utils/schema-v4-generator.js

```diff
@@ -100,7 +99,9 @@ class SchemaV4Generator {
     const schemaType = this.getSchemaTypeFor(baseObject);
 
     const hasKeys =
-      type.object(baseObject) && Object.keys(baseObject).length > 0;
+      baseObject !== null &&
```

Comment:
should this be `!==` or `!=`? (thinking of `undefined`)

Reply:
well `typeof undefined` is `'undefined'` so I guess it's sorted by the other line

Reply:
First checking for `null` to prevent false positive on the next line. Yes, then the next line is satisfied only on `object`. `undefined` value would reject then.